Formula/lammps.rb

```diff
@@ -35,41 +35,43 @@ class Lammps < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "f3877f35fa8ec630ff60c6cab01ebcc9ff8c9a7da3184618801b9316cd49c99a"
   end
 
+  depends_on "cmake" => :build
   depends_on "pkg-config" => :build
   depends_on "fftw"
   depends_on "gcc" # for gfortran
-  depends_on "jpeg"
+  depends_on "jpeg-turbo"
   depends_on "kim-api"
-  depends_on "libomp"
   depends_on "libpng"
   depends_on "open-mpi"
 
-  def install
-    ENV.cxx11
-
-    %w[serial mpi].each do |variant|
-      cd "src" do
-        system "make", "yes-all"
-        system "make", "no-lib"
-        system "make", "no-intel"
-        system "make", "yes-kim"
+  uses_from_macos "zlib"
 
-        system "make", variant,
-                       "LMP_INC=-DLAMMPS_GZIP",
-                       "FFT_INC=-DFFT_FFTW3 -I#{Formula["fftw"].opt_include}",
-                       "FFT_PATH=-L#{Formula["fftw"].opt_lib}",
-                       "FFT_LIB=-lfftw3",
-                       "JPG_INC=-DLAMMPS_JPEG -I#{Formula["jpeg"].opt_include} " \
-                       "-DLAMMPS_PNG -I#{Formula["libpng"].opt_include}",
-                       "JPG_PATH=-L#{Formula["jpeg"].opt_lib} -L#{Formula["libpng"].opt_lib}",
-                       "JPG_LIB=-ljpeg -lpng"
+  on_macos do
+    depends_on "libomp"
+  end
 
-        bin.install "lmp_#{variant}"
-        system "make", "clean-all"
-      end
+  def install
+    %w[serial mpi].each do |variant|
+      system "cmake", "-S", "cmake", "-B", "build_#{variant}",
+                      "-C", "cmake/presets/all_on.cmake",
+                      "-C", "cmake/presets/nolib.cmake",
+                      "-DPKG_INTEL=no",
+                      "-DPKG_KIM=yes",
+                      "-DLAMMPS_MACHINE=#{variant}",
+                      "-DBUILD_MPI=#{variant == "mpi" ? "yes" : "no"}",
+                      "-DBUILD_OMP=#{variant == "serial" ? "no" : "yes"}",
+                      "-DBUILD_SHARED_LIBS=yes",
+                      "-DFFT=FFTW3",
+                      "-DWITH_GZIP=yes",
+                      "-DWITH_JPEG=yes",
+                      "-DWITH_PNG=yes",
+                      "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                      *std_cmake_args
+      system "cmake", "--build", "build_#{variant}"
+      system "cmake", "--install", "build_#{variant}"
     end
 
-    pkgshare.install(%w[doc potentials tools bench examples])
+    pkgshare.install %w[doc tools bench examples]
```

Comment:
Just noting that I removed potentials since CMake installs it by default: https://github.com/lammps/lammps/blob/stable_23Jun2022/cmake/CMakeLists.txt#L758
```cmake
install(DIRECTORY ${LAMMPS_POTENTIALS_DIR} DESTINATION ${LAMMPS_INSTALL_DATADIR})
```